components/layout/block.rs

```diff
@@ -670,16 +670,16 @@ impl BlockFlow {
 
     /// Return true if this has a replaced fragment.
     ///
-    /// Text, Images, Inline Block and Canvas
+    /// Iframe, Images, SVG and Canvas
     /// (https://html.spec.whatwg.org/multipage/#replaced-elements) fragments are considered as
     /// replaced fragments.
     fn is_replaced_content(&self) -> bool {
         match self.fragment.specific {
-            SpecificFragmentInfo::ScannedText(_) |
             SpecificFragmentInfo::Svg(_) |
             SpecificFragmentInfo::Image(_) |
             SpecificFragmentInfo::Canvas(_) |
-            SpecificFragmentInfo::InlineBlock(_) => true,
+            SpecificFragmentInfo::Iframe(_) => true,
+            SpecificFragmentInfo::ScannedText(_) => panic!("A block that is also a text fragment!"),
```

Comment:
I thought I could complete this PR last night, but unfortunately ran into another bug: on replaced elements margins in block direction do not take effect. While this bug is really weird,  I will try to fix it this morning.